src/tools/tidy/src/tests_revision_unpaired_stdout_stderr.rs

```diff
@@ -84,7 +84,9 @@ pub fn check(tests_path: impl AsRef<Path>, bad: &mut bool) {
                 }
             });
 
-            let Some((test_name, _)) = test.to_str().map(|s| s.split_once('.')).flatten() else {
```

Comment:
You want https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.file_prefix which is sadly unstable

https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.with_extension should work too, and be nicer than... split 😭

Reply:
I *think* I actually want `file_stem` semantics here, because I want to keep the test name which is everything but the "rs" extension. This `split_once` doesn't seem right. E.g. `a.b.rs`, the test name is should be `a.b` not `a`. Revisioned output gets additional `.[rev].std{out,err}` after the test name to the best of my knowledge.

Reply:
> Could we not use some unstable features?

Tidy requires stable no?

Reply:
Oh 😢